locksmith/src/controllers/AirdropController.ts

```diff
@@ -0,0 +1,79 @@
+import { RequestHandler } from 'express'
+import Dispatcher from '../fulfillment/dispatcher'
+import { SubgraphService } from '@unlock-protocol/unlock-js'
+import normalizer from '../utils/normalizer'
+import { UserTokenMetadata } from '../models'
+import { sendEmail } from '../operations/wedlocksOperations'
+import config from '../config/config'
+
+export const createTransferCode: RequestHandler = async (request, response) => {
+  const lockAddress = normalizer.ethereumAddress(request.params.lockAddress)
+  const network = Number(request.params.network)
+  const keyId = request.params.keyId
+  const dispatcher = new Dispatcher()
+  const subgraph = new SubgraphService()
+
+  const user = await UserTokenMetadata.findOne({
+    where: {
+      tokenAddress: lockAddress,
+      chain: network,
+    },
+  })
+
+  const userData = user?.data?.userMetadata?.protected
+  const email = Object.keys(userData || {}).find((key) => {
+    return ['email', 'emailaddress', 'email_address', 'email-address'].includes(
+      key.toLowerCase()
+    )
+  })
+
+  if (!email) {
+    return response.status(404).send({
+      message: 'No email address found for this user',
+    })
+  }
+
+  const validPeriod = 60 * 60 * 15 // 15 minutes
+  const deadline = Math.floor(Date.now() / 1000) + validPeriod
+
+  const key = await subgraph.key(
+    {
+      where: {
+        tokenId: keyId.toLowerCase(),
+        lock: lockAddress.toLowerCase(),
+      },
+    },
+    {
+      network,
+    }
+  )
+
+  const owner = normalizer.ethereumAddress(key.owner)
+
+  const transfer = {
+    owner,
+    lock: lockAddress,
+    token: keyId,
+    deadline: deadline,
+  }
+
+  const transferCode = await dispatcher.createTransferCode(network, transfer)
```

Comment:
We may want to "reduce" the size of this string by using a Base64 number instead of the hex one?

Reply:
As communicated. The difference isn't huge and not worth the added hoop. I look for something which could be much smaller. 

For approximate reference:

Current: 140
Base36: 110
Base64: 90

Base64 also contain characters like `=/` . It's not something you can type. You will need to copy paste.